azurerm/internal/services/network/virtual_hub_connection_resource.go

```diff
@@ -55,16 +54,20 @@ func resourceArmVirtualHubConnection() *schema.Resource {
 				ValidateFunc: azure.ValidateResourceID,
 			},
 
+			// TODO 3.0: remove this property
 			"hub_to_vitual_network_traffic_allowed": {
-				Type:     schema.TypeBool,
-				Optional: true,
-				ForceNew: true,
+				Type:       schema.TypeBool,
+				Optional:   true,
+				ForceNew:   true,
+				Deprecated: "this property has been deprecated in the API and will be removed in version 3.0 of the provider",
```

Comment:
it'd be worth clarifying this as:

```suggestion
				Deprecated: "Due to a breaking behavioural change in the Azure API this property is no longer functional and will be removed in version 3.0 of the provider",
```